Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -788,6 +799,22 @@ private void Update()
                 UpdateRigHandles();
                 Target.transform.hasChanged = false;
             }
+            else if (HasBoundsOverrideChanged())
+            {
+                UpdateBounds();
+                UpdateRigHandles();
+            }
+        }
+        private bool HasBoundsOverrideChanged()
+        {
+            if (boundsOverride == null)
```

Comment:
Is it not supported to be able to set the boundsOverride to null? That is, it seems like it's possible to go like:

No bounds override -> bounds override A -> bounds override B -> bounds override A

But it looks like you can't be like:

No bounds override -> bounds override A -> No bounds override

This is fine if it's documented and there's no scenario in which that's useful. It's just generally good to be explicit about supported state changes (in the doccomments)

Reply:
Why can't you set boundsOverride to null? I have it in my test cases and it works fine. 

```
SetStatus("Remove target bounds override");
bbox.BoundsOverride = null;
Destroy(newObject);
newObject = null;
```

The resulting behavior is that the bounding box will snap to the bounds of the children.

I can add a comment that says "if you set boundsOverride to null, the bounding box will match the bounds of the game object or children.

Reply:
That's what I was trying to understand - if the bounds override is currently NOT null, then setting it to null will make HasBoundsOverrideChanged() report false (meaning, nothing has changed). However, in this example, you have changed something (you went from an override to no override)

I'm guessing that I'm missing something else (i.e. CreateRig gets called through some other path, or CreateRig doesn't need to be called when going from override to no override?)

Reply:
I'm not going to block on this one - I just want to understand the disconnect here between the supporting transitions to null and also making sure that we reevaluate state when transitioning to null.

Reply:
I'm not going to block on this one - I just want to understand the disconnect here between the supporting transitions to null and also making sure that we reevaluate state when transitioning to null.

Reply:
I see the same issue as will here - if I understand this correctly you should return true in this method in null case as well when the override was previously set.

Reply:
Thanks for the comments. I don't want to return true if boundsOverride is null because that would mean we update rig and handles every frame that boundsOverride == null. I think the name of the method "HasBoundsOverrideChanged" is confusing so I added a comment (I could also call it HasBoundsOverrideBoundsChanged but that seems really long). 

Can you have a look and see if this new method makes more sense to you? I verified the new method works with my test scene.